app/controllers/twitter_users_controller.rb

```diff
@@ -0,0 +1,79 @@
+class TwitterUsersController < ApplicationController
+  before_action :current_user
+
+  def subscribe
+    @screen_name = params[:screen_name]
+    if twitter_user_protected?
+      flash[:error] = "That user is protected; you cannot subscribe to their tweets."
+    else
+      if !twitter_user_exists?
+        # if twitter_user doesn't exist, create twitter_user and their tweets
+        create_twitter_user(@screen_name)
+        create_tweets(@twitter_user)
+      end
+      @twitter_user = TwitterUser.find_by(screen_name: @screen_name)
+      if @current_user.twitter_users.include?(@twitter_user)
```

Comment:
This is a great check. Making sure people can't subscribe twice to the same person :ok_hand: